scripts/misc/import_oldhgmd.pl

```diff
@@ -0,0 +1,596 @@
+
+    my $dump_mtmp = $old_dbhvar->prepare(qq[ SELECT  variation_feature_id, feature_stable_id, allele_string, consequence_types, cds_start, cds_end, cdna_start, cdna_end, translation_start, translation_end, distance_to_transcript from MTMP_transcript_variation where variation_feature_id in  (SELECT variation_feature_id from variation_feature where source_id = 8) ]);
+    open (my $mtmp, ">>$TMP_DIR/$old_mtmp_file") or die ("Cannot open $TMP_DIR/$old_mtmp_file: $!");
+    $dump_mtmp->execute();
+
+    while ( my $mtmpref = $dump_mtmp->fetchrow_arrayref() ){
+      my @mtmp = map {defined($_) ? $_: '\N'} @$mtmpref;
+      print $mtmp join("\t", @mtmp), "\n";
+    }
+
+    $dump_mtmp->finish();
+  }
+ 
+}
+
+sub manipulate_var_ids {
+  my $dbhvar = shift; 
+
+  my $select_max_var = $dbhvar->prepare(qq[SELECT MAX(variation_id) from variation]);
+  $select_max_var->execute();
+  my $var = $select_max_var->fetchall_arrayref();
+  my $max_var = $var->[0]->[0];
+  $max_var = $max_var + 1;
+  
+  my $select_max_feat = $dbhvar->prepare(qq[SELECT MAX(variation_feature_id) from variation_feature]);
+  $select_max_feat->execute();
+  my $vf = $select_max_feat->fetchall_arrayref();
+  my $max_var_feat = $vf->[0]->[0];
+  $max_var_feat = $max_var_feat + 1;
+  
+  $select_max_var->finish();
+  $select_max_feat->finish();
+ 
+  open(my $fh, '<', "$TMP_DIR/$sorted_old_var") or die "Cannot open file: $!";
+  open(my $new_var_file, '>', "$TMP_DIR/$new_var") or die "Cannot open file: $!";
+  while ( my $line = <$fh> ) {
+    chomp $line;
+    my @columns = split("\t", $line);
+
+    $columns[0] = $max_var++;
+    print $new_var_file join("\t", @columns), "\n";
+  }
+  close $fh;
+  close $new_var_file;
+  
+  $max_var = $var->[0]->[0];
+  $max_var = $max_var + 1;
+
+  open(my $old_var, '<', "$TMP_DIR/$sorted_old_var_feat") or die "Cannot open file: $!";
+  open(my $new_vf, '>', "$TMP_DIR/$new_var_feat") or die "Cannot open file: $!";
+
+  while (my $var_feat = <$old_var> ) {
+    chomp $var_feat;
+    my @columns = split ("\t", $var_feat);
+    
+    $columns[0] = $max_var_feat++;
+    $columns[5] = $max_var++;
+    print $new_vf join ("\t", @columns), "\n";
+  }
+
+  close $old_var;
+  close $new_vf;
+  
+  if ($config->{transcript}) {
+
+    my $select_max_tv = $dbhvar->prepare(qq[SELECT MAX(transcript_variation_id) from transcript_variation]);
+    $select_max_tv->execute();
+    my $tv = $select_max_tv->fetchall_arrayref();
+    my $max_tv = $tv->[0]->[0];
+    $max_tv = $max_tv + 1;
+
+    open($old_var, '<', "$TMP_DIR/$sorted_old_var_feat") or die "Cannot open file: $!";
+    open($new_vf, '<', "$TMP_DIR/$new_var_feat") or die "Cannot open file: $!";
+    
+    my @old_vid;
+    my @new_vid;
+    while (<$old_var>) {
+      chomp;
+      my $key = (split)[0];
+      push @old_vid, $key;
+    }
+    while (<$new_vf>) {
+      chomp;
+      my $new_vid = (split)[0];
+      push @new_vid, $new_vid;
+    }
+    close $old_var;
+    close $new_vf;
```

Comment:
This part can be improved in the future.
The new_ids -> old_ids can be assigned before at L280

Reply:
This part of the code needs to be updated. 
In some Perl versions, the order of an array is not always the same, so you're going to match wrong ids in some versions. 
It would be best if you created a hash instead.